### PR TITLE
Metal: Ensure `texture_create_from_extension` returns correct pixel format

### PR DESCRIPTION
**Pull request by stuartcarnie**
Prior implementation was ignoring the pixel format and returning the same texture.

Helps #98872

### CODE REVIEW

**Review by BastiaanOlij — APPROVED** [review]
The change looks good, I do wonder if we should validate the other parameters somehow to catch any issues but I think that is overkill.

Haven't been able to reproduce the OpenXR extension working on my Macbook Air but I am assuming that is an issue with the OpenXR PR, not with this. So I suggest this is merged ASAP.

File: drivers/metal/rendering_device_driver_metal.mm [hunk 1]
```diff
@@ -358,11 +358,28 @@ _FORCE_INLINE_ MTLSize mipmapLevelSizeFromSize(MTLSize p_size, NSUInteger p_leve
...
+				MTLTextureSwizzleRed,
+				MTLTextureSwizzleGreen,
+				MTLTextureSwizzleBlue,
+				MTLTextureSwizzleAlpha);
+		res = [src_texture newTextureViewWithPixelFormat:format
+											 textureType:src_texture.textureType
+												  levels:NSMakeRange(0, src_texture.mipmapLevelCount)
+												  slices:NSMakeRange(0, p_array_layers)
+												 swizzle:swizzle];
+		ERR_FAIL_NULL_V_MSG(res, TextureID(), "Unable to create texture view");
```
Comment (AThousandShips) [inline drivers/metal/rendering_device_driver_metal.mm:377]:
```suggestion
		ERR_FAIL_NULL_V_MSG(res, TextureID(), "Unable to create texture view.");
```

**Review by dsnopek — APPROVED** [review]
This is working for me as part of PR https://github.com/godotengine/godot/pull/98872, using Meta's XR Simulator on my Mac Mini

File: drivers/metal/rendering_device_driver_metal.mm (hunk 1 above)
Comment (stuartcarnie) [reply drivers/metal/rendering_device_driver_metal.mm]: Done

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**